wolfcrypt/src/sakke.c

```diff
@@ -0,0 +1,6795 @@
+/* sakke.c
+ *
+ * Copyright (C) 2006-2021 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+
+
+#ifdef HAVE_CONFIG_H
+    #include <config.h>
+#endif
+
+#include <wolfssl/wolfcrypt/settings.h>
+
+#ifdef NO_INLINE
+    #include <wolfssl/wolfcrypt/misc.h>
+#else
+    #define WOLFSSL_MISC_INCLUDED
+    #include <wolfcrypt/src/misc.c>
+#endif
+
+#ifdef WOLFSSL_HAVE_SP_ECC
+    #include <wolfssl/wolfcrypt/sp.h>
+#endif
+
+#ifdef WOLFCRYPT_HAVE_SAKKE
+
+#include <wolfssl/wolfcrypt/error-crypt.h>
+#include <wolfssl/wolfcrypt/sakke.h>
+#include <wolfssl/wolfcrypt/asn_public.h>
+
+/* SAKKE Build Options:
+ * WOLFSSL_SAKKE_SMALL:          Small code size version of SAKKE.
+ * WOLFSSL_SAKKE_SMALL_MODEXP:   Small code size for just SAKKE modexp.
+ */
+
+
+#ifdef WOLFCRYPT_SAKKE_CLIENT
+/*
+ * Initialize the client components of the SAKKE key.
+ *
+ * @param  [in]  key      SAKKE key to initialize.
+ * @return  0 on success.
+ * @return  MEMORY_E when dynamic memory allocation fails.
+ */
+static int sakke_init_client(SakkeKey* key)
+{
+    int err = 0;
+
+    key->tmp.p1 = wc_ecc_new_point_h(key->ecc.heap);
```

Comment:
NULL check on key, or is this done already by all callers?

Reply:
One caller and it checks key.